docs/source/pages/tutorials/detect_malicious_package.rst

```diff
@@ -0,0 +1,186 @@
+
+
+'''''''''''''''''''''''''''''''''''''
+Analyzing django without dependencies
+'''''''''''''''''''''''''''''''''''''
+
+First, we need to run the ``analyze`` command of Macaron to run a number of :ref:`checks <checks>` on the ``django`` package. In this tutorial, we are interested in the results of the ``mcn_detect_malicious_metadata_1``. Check :ref:`this tutorial <include_exclude_checks>` if you would like to exclude other checks.
+
+.. code-block:: shell
+
+  ./run_macaron.sh analyze -purl pkg:pypi/django@5.0.6 --skip-deps
+
+.. note:: By default, Macaron clones the repositories and creates output files under the ``output`` directory. To understand the structure of this directory please see :ref:`Output Files Guide <output_files_guide>`.
+
+.. code-block:: shell
+
+  open output/reports/pypi/django/django.html
+
+.. _fig_django-malware-check:
+
+.. figure:: ../../_static/images/tutorial_django_5.0.6_detect_malicious_metadata_check.png
+   :alt: Check ``mcn_detect_malicious_metadata_1`` result for ``django@5.0.6``
+   :align: center
+
+|
+
+The image above shows the result of the ``mcn_detect_malicious_metadata_1`` check for ``django@5.0.6``. The check has passed, which means this package is not malicious. If a package is malicious, this check fails, and if the ecosystem is not supported, the check returns ``UKNOWN``. You can also see the result of individual heuristics applied in this check under the ``Justification`` column.
+
+Now we can write a policy to enforce that all versions of ``django`` pass the ``mcn_detect_malicious_metadata_1`` check. The policy will be enforced against the ``analyze`` command run outputs cached in the local database at ``output/macaron.db``.
+
+.. code-block:: shell
+
+  ./run_macaron.sh verify-policy --database output/macaron.db --file policy.dl
+
+Where the policy looks like below:
+
+.. code-block:: prolog
+
+  #include "prelude.dl"
+
+  Policy("check-django", component_id, "Check django artifacts.") :-
+      check_passed(component_id, "mcn_detect_malicious_metadata_1").
+
+
+  apply_policy_to("check-django", component_id) :-
+      is_component(component_id, purl),
+      match("pkg:pypi/django@.*", purl).
+
+The ``match`` constraint in this policy allows us to apply the policy on all versions of ``django``. The result of this command should show that the policy succeeds with a zero exit code:
+
+.. code-block:: javascript
+
+    passed_policies
+        ['check-django']
+    component_satisfies_policy
+        ['1', 'pkg:pypi/django@5.0.6', 'check-django']
+    failed_policies
+    component_violates_policy
+
+Note that the ``match`` constraint applies a regex and can be expanded to enforce the ``mcn_detect_malicious_metadata_1`` check to pass on all Python packages analyzed so far by Macaron:
+
+.. code-block:: prolog
+
+  apply_policy_to("check-django", component_id) :-
```

Comment:
I think in this case, we should rename the policy to `check-all-pypi-packages` or something to prevent confusion:
```
.. code-block:: prolog
  #include "prelude.dl"
  Policy("check-all-pypi-packages", component_id, "Check django artifacts.") :-
      check_passed(component_id, "mcn_detect_malicious_metadata_1").
  apply_policy_to("check-all-pypi-packages", component_id) :-
      is_component(component_id, purl),
      match("pkg:pypi.*", purl).
```

Reply:
This modification is suggested as a variation of the previous policy and the policy name should not change.